### DISCUSSION TITLE
File mappings vsCode for debugging

### DISCUSSION THREAD

**Discussion by jacekidziak** [microsoft/vscode-cpptools#7443] [Q&A] [answered]
I get:
"Unable to open 'main.c': Unable to read file 'c:\mnt\c\xyz\common\main.c' (Error: Unable to resolve non-existing file 'c:\mnt\c\xyz\common\main.c').".
The file exists in c:\xyz\common\main.c.

The problem occurs when program hits breakpoint in any file.
It happens because i'm building project on WSL but debugging from windows through ssh and paths differs. I hoped to make it work with "pathSubstitutions" but it seems that it doesn't work?

Using ShowDevDebugOutput attribute im getting:
"["fullname","/mnt/c/xyz/common/main.c"]" as one of the last record.

Is there any way to make it work?

**Comment by sean-mcmanus** [comment]
Is this WSL1? It's recommended to use WSL2 with files existing under WSL (e.g. under /home/user instead of /mnt/c).

**Reply by jacekidziak** [reply]
Debugger itself works and sets breakpoints and tracks call stack correctly. It seems that the only thing that I need is to somehow tell vsCode/cpptools to correctly translate linux path that comes from debugger ("/mnt/c/xyz/common/main.c") to windows path where the file are located in host OS ( "c:\xyz\common\main.c"). Does the upgrade to WSL2 can help to solve the problem?

**Comment by jacekidziak** [comment, marked as answer]
I found the problem. 
Configuration of ssh part of launch.json used link in the cwd:
"            "ssh": {
                ...
                "cwd": "/home/user/<<link>>/common/",
                ...
            }"
Link leads to mnt\c\xyz.
After changing cwd to absolute path on WSL it started to work:
"cwd": "mnt/c/xyz/common/".
No path substitution is necessary right now.

Thank you for support.
